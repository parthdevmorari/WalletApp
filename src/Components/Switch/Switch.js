import React from 'react';
import { TouchableOpacity, Image } from 'react-native';
import styles from './SwitchStyle'
import * as resources from 'resources';

const Switch = ({ value, onPress, textStyle, size = 50, ...props }) => (
	<TouchableOpacity onPress={onPress} {...props}>
		<Image style={[styles.image,value===true?styles.active:styles.inActive]} source={value===true?resources.ON:resources.OFF} />
	</TouchableOpacity>
);

export default Switch;
