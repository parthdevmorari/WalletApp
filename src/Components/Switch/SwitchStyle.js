import { StyleSheet, Dimensions, StatusBar } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { color } from "../../Constants/Color";

const width = Dimensions.get('window').width;
const ITEM_WIDTH = Math.round(width - 70);
const styles = StyleSheet.create({
	image: {
        width: 50, height: 25, resizeMode:'contain', 	
	},
	inActive:{
	},
	active: {
		
	}

});
export default styles;

