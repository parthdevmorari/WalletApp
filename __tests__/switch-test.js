import 'react-native';
import React from 'react';
import Switch from '../src/Components/Switch/Switch'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('Switch Component', () => {
  test('Switch Component', () => {
    const wrapper = shallow(<Switch />);
    expect(wrapper.exists()).toBe(true);
  });
  it('Switch  toMatchSnapshot', () => {
    const wrapper = shallow(<Switch />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});