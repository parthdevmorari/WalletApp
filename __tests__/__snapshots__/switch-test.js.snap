// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component Switch  toMatchSnapshot 1`] = `
<TouchableOpacity>
  <Image
    source={
      Object {
        "testUri": "../../../src/resources/off.png",
      }
    }
    style={
      Array [
        Object {
          "height": 25,
          "resizeMode": "contain",
          "width": 50,
        },
        Object {},
      ]
    }
  />
</TouchableOpacity>
`;
